app/policies/need_policy.rb

```diff
@@ -46,6 +46,20 @@ def show?
     Pundit.policy_scope!(@user, Need).where(id: @record.id).exists?
   end
 
+  def destroy?
+    # managers can always destroy
+    return true if admin?
+
+    # check ownership of need record
+    need = Pundit.policy_scope!(@user, Need).where(id: @record.id)
```

Comment:
Policy scope for certain roles (MDT, Food Hub, Council Services), restricts viewable needs to those that are assigned to them, their team, or their role.

This filtering is done by the policy scope - so `Pundit.policy_scope!(@user, Need).where(id: @record.id).exists?` will return false if it is not assigned to one of those three that put it in the policy scope, even if the user created it.

Reply:
ok I get it now, thanks

Reply:
@jamiebuckley  amended now